test/spec/modules/ttdBidAdapter_spec.js

```diff
@@ -0,0 +1,1090 @@
+    clonedBidderRequest.bids = bidRequests;
+    return spec.buildRequests(bidRequests, clonedBidderRequest);
+  }
+
+  describe('isBidRequestValid', function() {
+    function makeBid() {
+      return {
+        'bidder': 'ttd',
+        'params': {
+          'supplySourceId': 'supplier',
+          'publisherId': '22222222',
+          'placementId': 'some-PlacementId_1',
+          'siteId': 'testSiteId'
+        },
+        'mediaTypes': {
+          'banner': {
+            'sizes': [
+              [300, 250]
+            ]
+          }
+        },
+        'adUnitCode': 'adunit-code',
+        'bidId': '30b31c1838de1e',
+        'bidderRequestId': '22edbae2733bf6',
+        'auctionId': '1d1a030790a475',
+      };
+    }
+
+    describe('core', function () {
+      it('should return true when required params found', function () {
+        expect(spec.isBidRequestValid(makeBid())).to.equal(true);
+      });
+
+      it('should return false when publisherId not passed', function () {
+        let bid = makeBid();
+        delete bid.params.publisherId;
+        expect(spec.isBidRequestValid(bid)).to.equal(false);
+      });
+
+      it('should return false when supplySourceId not passed', function () {
+        let bid = makeBid();
+        delete bid.params.supplySourceId;
+        expect(spec.isBidRequestValid(bid)).to.equal(false);
+      });
+
+      it('should return false when publisherId is longer than 64 characters', function () {
+        let bid = makeBid();
+        bid.params.publisherId = '1111111111111111111111111111111111111111111111111111111111111111111111';
+        expect(spec.isBidRequestValid(bid)).to.equal(false);
+      });
+
+      it('should return false when siteId not passed', function () {
+        let bid = makeBid();
+        delete bid.params.siteId;
+        expect(spec.isBidRequestValid(bid)).to.equal(false);
+      });
+
+      it('should return false when siteId is longer than 50 characters', function () {
+        let bid = makeBid();
+        bid.params.siteId = '1111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111'
+        expect(spec.isBidRequestValid(bid)).to.equal(false);
+      });
+
+      it('should return false when placementId not passed', function () {
+        let bid = makeBid();
+        delete bid.params.placementId;
+        expect(spec.isBidRequestValid(bid)).to.equal(false);
+      });
+
+      it('should return false when the placementId is longer than 128 characters', function () {
+        let bid = makeBid();
+        bid.params.placementId = '1111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111'; // 130 characters
+        expect(spec.isBidRequestValid(bid)).to.equal(false);
+      });
+
+      it('should return false if categories is passed in correctly', function () {
```

Comment:
should return true?

Reply:
test has been removed as `categories` has been removed as a param